ocaml/xapi/xapi.ml

```diff
@@ -609,6 +611,32 @@ let check_network_reset () =
           else
             ""
         in
+        let mode_v6 =
```

Comment:
This could be made simpler using `let*` but it's a matter of taste and being used to it. Below I'm demonstrating the use of the `Option.bind` monad.

```
let args = [ "MODE_V6", "autoconf" ]

let f args =
  let ( let* ) = Option.bind in
  let* mode = List.assoc_opt "MODE_V6" args in
  match mode with 
  | "static" -> Some `Static
  | "autoconf" -> Some `Autoconf
  | "dhcp" | _ -> Some `DHCP
```